src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -74,56 +75,65 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
 
                 var span = new TextSpan(position, 0);
                 var semanticModel = await document.GetSemanticModelForSpanAsync(span, cancellationToken).ConfigureAwait(false);
-                var type = typeInferenceService.InferType(semanticModel, position,
-                    objectAsDefault: true,
+                var types = typeInferenceService.InferTypes(semanticModel, position,
                     cancellationToken: cancellationToken);
 
-                // If we have a Nullable<T>, unwrap it.
-                if (type.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T)
+                if (types.Length == 0)
                 {
-                    type = type.GetTypeArguments().FirstOrDefault();
-
-                    if (type == null)
-                    {
-                        return;
-                    }
+                    types = ImmutableArray.Create<ITypeSymbol>(semanticModel.Compilation.ObjectType);
                 }
 
-                if (type.TypeKind != TypeKind.Enum)
+                foreach (var typeIterator in types)
                 {
-                    type = TryGetEnumTypeInEnumInitializer(semanticModel, token, type, cancellationToken) ??
-                           TryGetCompletionListType(type, semanticModel.GetEnclosingNamedType(position, cancellationToken), semanticModel.Compilation);
+                    var type = typeIterator;
 
-                    if (type == null)
+                    // If we have a Nullable<T>, unwrap it.
+                    if (type.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T)
                     {
-                        return;
+                        type = type.GetTypeArguments().FirstOrDefault();
+
+                        if (type == null)
+                        {
+                            continue;
+                        }
                     }
-                }
 
-                if (!type.IsEditorBrowsable(options.GetOption(CompletionOptions.HideAdvancedMembers, semanticModel.Language), semanticModel.Compilation))
-                {
-                    return;
-                }
+                    if (type.TypeKind != TypeKind.Enum)
+                    {
+                        type = TryGetEnumTypeInEnumInitializer(semanticModel, token, type, cancellationToken) ??
```

Comment:
I'm a little confused by this logic. If the type is not an enum, we try to get an enum?

Reply:
There can be multiple types. Some of them can be enums and some are not. The scenarios we're fixing is an overload with enum in one case and not enum in another.
We should bail out the non-enum case but provide results for the enum one. It should not depend which one was taken with FirstOrDefault.

---
In reply to: [292626711](https://github.com/dotnet/roslyn/pull/36293#discussion_r292626711) [](ancestors = 292626711)